src/main/java/com/hierynomus/smbj/share/NamedPipe.java

```diff
@@ -141,7 +196,15 @@ public int transact(byte[] inBuffer, byte[] outBuffer) {
      * @return the number of bytes written to <code>outBuffer</code>
      */
     public int transact(byte[] inBuffer, int inOffset, int inLength, byte[] outBuffer, int outOffset, int outLength) {
-        return ioctl(FSCTL_PIPE_TRANSCEIVE, true, inBuffer, inOffset, inLength, outBuffer, outOffset, outLength);
+        if (outBuffer.length - outOffset < outLength) {
+            throw new IllegalArgumentException();
+        }
+        final byte[] transactOutBuffer = transact(inBuffer);
+        if (transactOutBuffer.length > outLength) {
+            throw new BufferOverflowException();
```

Comment:
Just getting the single user use case working correctly is probably ok. Imagine you do a 'create user' RPC call. You try once, get buffer overflow. Try again and then get user already exists. The operation was performed, we just threw away the answer. Would be kind of weird IMO.